libs/automation/actionBuilders/textInputLogicBuilder.js

```diff
@@ -72,76 +72,59 @@ class TextInputLogicBuilder extends LogicBuilder
      * @param {object} page puppeteer page
      */
     perform = async (action, target, page) => {
-        const { insertScripts } = this.meta;
-
-        await addXhrListener(page);
-        await addNavigationListener(page);
-
-        await this.clearInputField(target.selector, page);
-        await page.type(target.selector, target.input, { delay: 300 });
-        await page.waitForNetworkIdle();
-
-        await Promise.all([
-            awaitXhrResponse(),
-            awaitNavigation(),
-            page.waitForTimeout(1000),
-        ]);
-
-        await this.pressKeys(target.keyPresses, page);
-        await page.waitForNetworkIdle();
-
-        await Promise.all([
-            awaitXhrResponse(),
-            awaitNavigation(),
-            page.waitForTimeout(1000),
-        ]);
-
-        await insertScripts(page);
-        
-        await removeXhrListener();
-        removeNavigationListener(); 
+        try{
+            const inputField = await page.locator(target.selector).first();
+            await inputField.fill('');  // .fill() waits for actionability
+            await inputField.type(target.input, { delay: 200 });
+            await this.pressKeys(target.keyPresses, inputField);
+            await page.waitForLoadState('networkidle');  // TODO: ADD DYNAMIC TIMEOUT
```

Comment:
Wait for timeout instead and check if the navigations are reliable. If yes, add xhrHandler back